opal/mca/accelerator/accelerator.h

```diff
@@ -309,6 +309,71 @@ typedef int (*opal_accelerator_base_module_mem_release_fn_t)(
 typedef int (*opal_accelerator_base_module_get_address_range_fn_t)(
     int dev_id, const void *ptr, void **base, size_t *size);
 
+/**
+ * Queries whether the device supports Inter Process Communication
+ * or not. If true, the functions:
+ *
+ * opal_accelerator_base_module_get_ipc_handle_size_fn_t()
+ * opal_accelerator_base_module_get_ipc_handle_fn_t()
+ * opal_accelerator_base_module_open_ipc_handle_fn_t()
+ * opal_accelerator_base_module_close_ipc_handle_fn_t()
+ *
+ * must be implemented.
+ *
+ * @return true              IPC supported
+ * @return false             IPC not supported
+ */
+typedef bool (*opal_accelerator_base_module_is_ipc_enabled_fn_t)(void);
+
+/**
+ * Returns the size of the Inter Process Communication memory handle
+ *
+ * @return size_t            Size of the IPC memory handle
+ */
+typedef size_t (*opal_accelerator_base_module_get_ipc_handle_size_fn_t)(void);
+
+/**
+ * Gets an interprocess memory handle for an existing device memory allocation.
+ *
+ * @param[IN] dev_id         Associated device for the IPC memory handle or
+ *                           MCA_ACCELERATOR_NO_DEVICE_ID
+ * @param[IN] dev_ptr        Base pointer to previously allocated device
+ *                           memory
+ * @param[OUT] handle        Pointer to user allocated mem handle to return
+ *                           the memory handle in.
+ *
+ * @return                   OPAL_SUCCESS or error status on failure
+ *
+ */
+typedef int (*opal_accelerator_base_module_get_ipc_handle_fn_t)(
+    int dev_id, void *dev_ptr, void **handle);
```

Comment:
Why a void* instead of a class object like the rest of the interface?

Reply:
New revision encapsulates `handle` and `size`, therefore we don't need the API to get handle size.

However, I think it's necessary to keep device id and base address - didn't find a simplistic way to hide them, maybe in IPC cache if we decide to implement that.